src/components/SelectedSurveyPage.jsx

```diff
@@ -26,6 +26,19 @@ export default class SelectedSurveyPage extends PureComponent {
     fetchSurvey(id);
   }
 
+  // TODO(ayushbhagat): Check for cycles.
```

Comment:
A bit off-topic, but will this be just front-end validation ?
It'll prolly be harder to do it in the back-end as well, but we should discuss ?